eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -294,7 +295,11 @@ public function findTrashItems(Query $query)
 
         $trashItems = array();
         foreach ($spiTrashItems as $spiTrashItem) {
-            $trashItems[] = $this->buildDomainTrashItemObject($spiTrashItem);
+            try {
+                $trashItems[] = $this->buildDomainTrashItemObject($spiTrashItem);
+            } catch (UnauthorizedException $e) {
+                // Do nothing, thus exclude items the current user doesn't have read access to.
```

Comment:
Makes sense, but maybe we can log here something about user not being able to read TrashItem?

Reply:
Is there any use case for that? It's conceptually the same as when some content under Home is not readable, that is also afaik silently hidden.

Reply:
I'd say it gives a maintainer easy way to answer customer questions like: "when I trash item I don't see any details about it, but my colleague sees something different doing the same thing, why?". 
But maybe I'm overthinking this :laughing:

Reply:
We don't log this in other places, so I agree with @glye here, but maybe trash interface should clearly state somewhere it only shows the items user have access to.

Reply:
Yes, that's true. Given the currently default Editor role, which means that editors can't see things they themselves put in trash, it is understandable that they'd complain that the item they trashed has disappeared. I'll see if I can add a notice in the UI.

Reply:
@glye For that you might need to #2125 to return different http status code when user no longer has access to what they have moved using REST then, and not just a mention of it in spec. or?

Reply:
@andrerom Right now it's mentioned in docblock, and I have a fixed notice in the UI, see https://github.com/ezsystems/PlatformUIBundle/pull/921
I won't be too disappointed if that isn't kosher, don't much like it myself :)

Reply:
@andrerom Self-correction: What status code the trashing rest call returns is not relevant for this PR. Here I have a static message: "This list includes only content that you have read access to." It could be enhanced to only show that message when there are such unreadable items in trash, for example by fetching (count) with sudo in addition to the current fetch and compare the number of items.

In https://github.com/ezsystems/ezpublish-kernel/pull/2125 we could use such a modified status code to tell the user that the item is trashed and he no longer has access to it - but should we? Such a message (orange warning) flashing to the user could just add confusion.